src/dev/impl/DevToys/ViewModels/Tools/Graphic/ImageConverter/ImageConverterToolViewModel.cs

```diff
@@ -0,0 +1,290 @@
+
+        #region SelectFilesBrowseCommand
+
+        public IAsyncRelayCommand SelectFilesBrowseCommand { get; }
+
+        private async Task ExecuteSelectFilesBrowseCommandAsync()
+        {
+            try
+            {
+                await ThreadHelper.RunOnUIThreadAsync(async () =>
+                {
+                    HasInvalidFilesSelected = false;
+
+                    var filePicker = new FileOpenPicker
+                    {
+                        ViewMode = PickerViewMode.List,
+                        SuggestedStartLocation = PickerLocationId.ComputerFolder
+                    };
+
+                    for (int i = 0; i < SupportedFileExtensions.Length; i++)
+                    {
+                        filePicker.FileTypeFilter.Add(SupportedFileExtensions[i]);
+                    }
+
+                    IReadOnlyList<StorageFile> files = await filePicker.PickMultipleFilesAsync();
+                    for (int i = 0; i < files.Count; i++)
+                    {
+                        QueueNewConversionWorkItem(files[i]);
+                    }
+                });
+            }
+            catch (Exception) { IsInfoBarOpen = true; }
+        }
+
+        #endregion
+
+        #region DeleteAllCommand
+
+        public IRelayCommand DeleteAllCommand { get; }
+
+        private void ExecuteDeleteAllCommand()
+        {
+            ConversionWorkQueue.Clear();
+        }
+
+        #endregion
+
+        #region SaveAllCommand
+
+        public IAsyncRelayCommand SaveAllCommand { get; }
+
+        private async Task ExecuteSaveAllCommandAsync()
+        {
+            try
+            {
+                var works = ConversionWorkQueue.ToList();
+
+                var folderPicker = new FolderPicker
+                {
+                    ViewMode = PickerViewMode.List
+                };
+
+                StorageFolder? selectedFolder = await folderPicker.PickSingleFolderAsync();
+
+                if (selectedFolder is not null)
+                {
+                    ICollection<Task> conversionTasks = new List<Task>();
+
+                    foreach (ImageConversionWorkItem work in works)
+                    {
+                        conversionTasks.Add(SaveConversionWorkItem(selectedFolder, work));
+                    }
+
+                    await Task.WhenAll(conversionTasks);
+                }
+            }
+            catch (Exception) { IsInfoBarOpen = true; }
+        }
+
+        #endregion
+
+        #region CloseInfoBarButtonCommand
+
+        public IRelayCommand CloseInfoBarButtonCommand { get; }
+
+        private void ExecuteCloseInfoBarButtonCommand()
+        {
+            IsInfoBarOpen = false;
+        }
+
+        #endregion
+
+        private async Task SaveConversionWorkItem(StorageFolder selectedFolder, ImageConversionWorkItem work)
+        {
+            StorageFile newFile = await selectedFolder.CreateFileAsync(work.FileName, CreationCollisionOption.ReplaceExisting);
```

Comment:
I tried to bug bash a little bit the feature and I think I found an issue here :)

Scenario :
1. Add a file `C:\folder\foo.jpeg` to convert to PNG.
2. Click `Save All` and save in `C:\folder`.
3. Line 259 crash because the file `foo.jpeg` is already opened.

I looked a little bit at where is the file opened, and it appears that it's due to the `ImageConversionWorkItem.Bitmap` property. That property is non-null at this moment and it seems like it keeps the file opened even is you correctly dispose the `IRandomAccessStream` in `DecodeImageAsync`.

In addition to this problem, I realized that the SoftwareBitmap is never disposed, meaning that we keep the bitmap indefinitely in memory (and therefore, memory leak). I just gave it a try and confirmed my thought:
1. Select 100 images in the tool.
2. Observe in Task Manager that the RAM increased a lot.
3. Click `Delete All` without doing the conversion.
4. All items disappear from the UI, but the RAM is still high in Task Manager.

Therefore, here is what I suggest.
1. Between the moment the user selects a file and the moment he clicks `Save`, minutes or even hours can elapse. Meanwhile, it seems like we don't need that SoftwareBitmap until the user Save the file. Therefore, as a first step, I'd suggest to refactor the code so DecodeImageAsync is only called in the Save/Conversion process.
2. As part of this refactoring,  DecodeImageAsync could return the SoftwareBitmap instead of using the Bitmap property. That would allow to have a `using (var bitmap = await DecodeImageAsync)`, so the resource can get disposed ASAP, and we can remove the `Bitmap` property.
3. Lines 261 to 268 are pretty much the same in ImageConversionWorkItem. Let's move this code in a new async method (off UI thread) in ImageConversionWorkItem so we avoid duplicating important piece of code.

Finally, I'm suspecting another issue with lines 259 and 260. Let's take the same scenario as above:
1. Add a file `C:\folder\foo.jpeg` to convert to PNG.
2. Click `Save All` and save in `C:\folder`.
Wouldn't line 259 **replace** `foo.jpeg` even if we don't write in the file? That doesn't sound right.

Reply:
> Finally, I'm suspecting another issue with lines 259 and 260. Let's take the same scenario as above:
> 1. Add a file C:\folder\foo.jpeg to convert to PNG.
> 2. Click Save All and save in C:\folder.
> Wouldn't line 259 replace foo.jpeg even if we don't write in the file? That doesn't sound right.

About this, I've used the same approach that we're currently using in PngJpgCompressor tool. Anyway, answering your  comment I think it won´t be replaced because file extensions are different.

Reply:
On the other hand, about first issue:

> Between the moment the user selects a file and the moment he clicks Save, minutes or even hours can elapse. Meanwhile, it seems like we don't need that SoftwareBitmap until the user Save the file. Therefore, as a first step, I'd suggest to refactor the code so DecodeImageAsync is only called in the Save/Conversion process.

I think you're right, we're not diposing the Bitmap property but with the approach you've proposed we will need a StorageFile type property to decode the image when needed so will occur the same. My suggestion to fix point 2 and 3 is implement IDisposable in ```ImageConversionWorkItem``` class and dispose the Bitmap property if isn´t null. I've commited and pushed those changes so I would like to know your opinion 😄

Reply:
> Scenario :
> 
> 1. Add a file C:\folder\foo.jpeg to convert to PNG.
> 2. Click Save All and save in C:\folder.
> 3. Line 259 crash because the file foo.jpeg is already opened.

Good catch! I have fixed this in [this commit](https://github.com/veler/DevToys/pull/139/commits/463854f3892b33d29a8391d5bf967172ea09f82a). This approach was also applied in JpgPngCompressor tool so it might be a good idea to review it, just in case.

Reply:
> > Finally, I'm suspecting another issue with lines 259 and 260. Let's take the same scenario as above:
> > 
> > 1. Add a file C:\folder\foo.jpeg to convert to PNG.
> > 2. Click Save All and save in C:\folder.
> >    Wouldn't line 259 replace foo.jpeg even if we don't write in the file? That doesn't sound right.
> 
> About this, I've used the same approach that we're currently using in PngJpgCompressor tool. Anyway, answering your comment I think it won´t be replaced because file extensions are different.

Can you please verify that it is the case? Because on line 259, we first create the file with the old name (foo.jpeg) at the location where the file exists already and we tell to replace the existing one. And only on the line after we rename it with a different extension. That is quite confusing to me.

> > Between the moment the user selects a file and the moment he clicks Save, minutes or even hours can elapse. Meanwhile, it seems like we don't need that SoftwareBitmap until the user Save the file. Therefore, as a first step, I'd suggest to refactor the code so DecodeImageAsync is only called in the Save/Conversion process.
> 
> I think you're right, we're not diposing the Bitmap property but with the approach you've proposed we will need a StorageFile type property to decode the image when needed so will occur the same. My suggestion to fix point 2 and 3 is implement IDisposable in `ImageConversionWorkItem` class and dispose the Bitmap property if isn´t null (before the garbage collection does that). I've commited and pushed those changes so I would like to know your opinion 😄

Yes, I've been thinking about using IDisposable, but there's still the fact that the bitmap will be loaded in memory for a while. I think it's best to load it the latest possible in this scenario, which is why I suggested my approach. :)

Reply:
> Can you please verify that it is the case? Because on line 259, we first create the file with the old name (foo.jpeg) at the location where the file exists already and we tell to replace the existing one. And only on the line after we rename it with a different extension. That is quite confusing to me.

Just changed that hahaha

Reply:
> Yes, I've been thinking about using IDisposable, but there's still the fact that the bitmap will be loaded in memory for a while. I think it's best to load it the latest possible in this scenario, which is why I suggested my approach. :)

Both Bitmap or StorageFile are needed to be stored in the WorkItem as a property, so if we're not using the Bitmap property approach we will need to store the StorageFile as a property to call the decode method when saving.

Reply:
> Both Bitmap or StorageFile are needed to be stored in the WorkItem as a property, so if we're not using the Bitmap property approach we will need to store the StorageFile as a property to call the decode method when saving 🤔.

That is fine. StorageFile is just a pointer to that file, it's not the actual file being loaded in memory. However. by keeping the StorageFile in the work item, we will need to check whether the file still exists on the hard drive when doing the conversion (after all, users could select a file in DevToys, delete the file from the hard drive, then click Save). :)

Feel free to jump on Discord for a quick chat if you want.

Reply:
Cool, so I have implemented that approach 😄